### ISSUE TITLE
route53 not being cleaned

### ISSUE THREAD

**Issue by cloin** [ansible/workshops#681]
##### SUMMARY

I've been meaning to look at this closely but keep getting distracted. On teardown events, only some r53 sets are being cleaned up. In regular playbook output, you can see something like this
```
TASK [code_server : delete dns entries for vs code for each student] ***********************
changed: [localhost] => (item=1)
changed: [localhost] => (item=2)
changed: [localhost] => (item=3)
skipping: [localhost] => (item=4)
skipping: [localhost] => (item=5)
skipping: [localhost] => (item=6)
skipping: [localhost] => (item=7)
skipping: [localhost] => (item=8)
skipping: [localhost] => (item=9)
changed: [localhost] => (item=10)
changed: [localhost] => (item=11)
```
##### ISSUE TYPE

 - Bug Report

**Comment by cloin** [comment]
> I bet the route53_facts is only returning the first 100 because of pagination....
> 
> I am pretty sure I had a bug open on this: https://github.com/ansible/ansible/issues/49457
> 
> I think this is usually not a problem b/c the workshop max is around 50, and it can handle 100, so if random entries get in there alphabetically somehow it will be OK....
> 
> but I think you might hit this issue at 100
> 
> and I think the -code prefix is now making this more complicated :-|

**Comment by cloin** [comment]
```
TASK [code_server : delete dns entries for vs code for each student] **********************************************************
task path: /Users/colin/projects/workshops/cloin-ansible-workshops/provisioner/roles/code_server/tasks/teardown.yml:16
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: colin
<localhost> EXEC /bin/sh -c 'echo ~colin && sleep 0'
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/colin/.ansible/tmp/ansible-tmp-1582050556.605056-19875695574896 `" && echo ansible-tmp-1582050556.605056-19875695574896="` echo /Users/colin/.ansible/tmp/ansible-tmp-1582050556.605056-19875695574896 `" ) && sleep 0'
Using module file /Users/colin/projects/ansible/stable-2.9/lib/ansible/modules/cloud/amazon/route53.py
<localhost> PUT /Users/colin/.ansible/tmp/ansible-local-3588y4y2j409/tmpiudstfi1 TO /Users/colin/.ansible/tmp/ansible-tmp-1582050556.605056-19875695574896/AnsiballZ_route53.py
<localhost> EXEC /bin/sh -c 'chmod u+x /Users/colin/.ansible/tmp/ansible-tmp-1582050556.605056-19875695574896/ /Users/colin/.ansible/tmp/ansible-tmp-1582050556.605056-19875695574896/AnsiballZ_route53.py && sleep 0'
<localhost> EXEC /bin/sh -c '/usr/bin/env python /Users/colin/.ansible/tmp/ansible-tmp-1582050556.605056-19875695574896/AnsiballZ_route53.py && sleep 0'
<localhost> EXEC /bin/sh -c 'rm -f -r /Users/colin/.ansible/tmp/ansible-tmp-1582050556.605056-19875695574896/ > /dev/null 2>&1 && sleep 0'
changed: [localhost] => (item=1) => changed=true
  ansible_loop_var: item
  invocation:
    module_args:
      alias: null
      alias_evaluate_target_health: false
      alias_hosted_zone_id: null
      aws_access_key: null
      aws_secret_key: null
      debug_botocore_endpoint_logs: false
      ec2_url: null
      failover: null
      health_check: null
      hosted_zone_id: null
      identifier: null
      overwrite: null
      private_zone: false
      profile: null
      record: student1-code.win-21820-1.rhdemo.io
      region: null
      retry_interval: 500
      security_token: null
      state: absent
      ttl: 3600
      type: A
      validate_certs: true
      value:
      - 3.80.126.3
      vpc_id: null
      wait: false
      wait_timeout: 300
      weight: null
      zone: rhdemo.io
  item: 1
skipping: [localhost] => (item=2)  => changed=false
  ansible_loop_var: item
  item: 2
  skip_reason: Conditional result was False
```

**Comment by cloin** [comment]
Hey @IPvSean, there's something in the logic for tearing down tower and code-server records that will always leave at least 1 record behind. 

https://github.com/ansible/workshops/blob/8ea41b25340f89f53d6392a91f1ff080a699f3ed/provisioner/roles/aws_dns/tasks/teardown.yml#L16-L27

**Comment by cloin** [comment]
In this teardown, 2 students had both code-server records removed but only student1's tower record was removed:

```
TASK [aws_dns : GRAB ZONE ID] *************************************************************************************************
ok: [localhost] => changed=false
  comment: ''
  delegation_set_id: ''
  invocation:
    module_args:
      aws_access_key: null
      aws_secret_key: null
      comment: ''
      debug_botocore_endpoint_logs: false
      delegation_set_id: null
      ec2_url: null
      hosted_zone_id: null
      profile: null
      region: null
      security_token: null
      state: present
      validate_certs: true
      vpc_id: null
      vpc_region: null
      zone: rhdemo.io
  name: rhdemo.io.
  private_zone: false
  result:
    comment: ''
    delegation_set_id: ''
    name: rhdemo.io.
    private_zone: false
    vpc_id: null
    vpc_region: null
    zone_id: Z1TYCDVCKYS6QZ
  vpc_id: null
  vpc_region: null
  zone_id: Z1TYCDVCKYS6QZ

TASK [aws_dns : GRAB ROUTE53 INFORMATION] *************************************************************************************
ok: [localhost] => changed=false
  IsTruncated: true
  MaxItems: '2'
  NextRecordName: student2.win-21820-1.rhdemo.io.
  NextRecordType: A
  ResourceRecordSets:
  - Name: student1.win-21820-1.rhdemo.io.
    ResourceRecords:
    - Value: 3.80.126.3
    TTL: 3600
    Type: A
  - Name: student2-code.win-21820-1.rhdemo.io.
    ResourceRecords:
    - Value: 34.228.61.238
    TTL: 3600
    Type: A
  ResponseMetadata:
    HTTPHeaders:
      content-length: '751'
      content-type: text/xml
      date: Tue, 18 Feb 2020 19:27:53 GMT
      x-amzn-requestid: 2f2428de-5761-4254-8c84-c4cbace94a51
    HTTPStatusCode: 200
    RequestId: 2f2428de-5761-4254-8c84-c4cbace94a51
    RetryAttempts: 0
  invocation:
    module_args:
      aws_access_key: null
      aws_secret_key: null
      change_id: null
      debug_botocore_endpoint_logs: false
      delegation_set_id: null
      dns_name: null
      ec2_url: null
      health_check_id: null
      health_check_method: list
      hosted_zone_id: Z1TYCDVCKYS6QZ
      hosted_zone_method: list
      max_items: '2'
      next_marker: null
      profile: null
      query: record_sets
      region: null
      resource_id: null
      security_token: null
      start_record_name: student1.win-21820-1.rhdemo.io
      type: A
      validate_certs: true

TASK [aws_dns : DELETE DNS ENTRIES FOR EACH STUDENT] **************************************************************************
changed: [localhost] => (item=1) => changed=true
  ansible_loop_var: item
  invocation:
    module_args:
      alias: null
      alias_evaluate_target_health: false
      alias_hosted_zone_id: null
      aws_access_key: null
      aws_secret_key: null
      debug_botocore_endpoint_logs: false
      ec2_url: null
      failover: null
      health_check: null
      hosted_zone_id: null
      identifier: null
      overwrite: null
      private_zone: false
      profile: null
      record: student1.win-21820-1.rhdemo.io
      region: null
      retry_interval: 500
      security_token: null
      state: absent
      ttl: 3600
      type: A
      validate_certs: true
      value:
      - 3.80.126.3
      vpc_id: null
      wait: false
      wait_timeout: 300
      weight: null
      zone: rhdemo.io
  item: 1
skipping: [localhost] => (item=2)  => changed=false
  ansible_loop_var: item
  item: 2
  skip_reason: Conditional result was False

TASK [code_server : GRAB ZONE ID] *********************************************************************************************
ok: [localhost] => changed=false
  comment: ''
  delegation_set_id: ''
  invocation:
    module_args:
      aws_access_key: null
      aws_secret_key: null
      comment: ''
      debug_botocore_endpoint_logs: false
      delegation_set_id: null
      ec2_url: null
      hosted_zone_id: null
      profile: null
      region: null
      security_token: null
      state: present
      validate_certs: true
      vpc_id: null
      vpc_region: null
      zone: rhdemo.io
  name: rhdemo.io.
  private_zone: false
  result:
    comment: ''
    delegation_set_id: ''
    name: rhdemo.io.
    private_zone: false
    vpc_id: null
    vpc_region: null
    zone_id: Z1TYCDVCKYS6QZ
  vpc_id: null
  vpc_region: null
  zone_id: Z1TYCDVCKYS6QZ

TASK [code_server : GRAB ROUTE53 INFORMATION] *********************************************************************************
ok: [localhost] => changed=false
  IsTruncated: true
  MaxItems: '2'
  NextRecordName: student2.win-21820-1.rhdemo.io.
  NextRecordType: A
  ResourceRecordSets:
  - Name: student1-code.win-21820-1.rhdemo.io.
    ResourceRecords:
    - Value: 3.80.126.3
    TTL: 3600
    Type: A
  - Name: student2-code.win-21820-1.rhdemo.io.
    ResourceRecords:
    - Value: 34.228.61.238
    TTL: 3600
    Type: A
  ResponseMetadata:
    HTTPHeaders:
      content-length: '756'
      content-type: text/xml
      date: Tue, 18 Feb 2020 19:27:56 GMT
      x-amzn-requestid: d2f4cb5f-6378-48e1-abaf-a0423eb700a5
    HTTPStatusCode: 200
    RequestId: d2f4cb5f-6378-48e1-abaf-a0423eb700a5
    RetryAttempts: 0
  invocation:
    module_args:
      aws_access_key: null
      aws_secret_key: null
      change_id: null
      debug_botocore_endpoint_logs: false
      delegation_set_id: null
      dns_name: null
      ec2_url: null
      health_check_id: null
      health_check_method: list
      hosted_zone_id: Z1TYCDVCKYS6QZ
      hosted_zone_method: list
      max_items: '2'
      next_marker: null
      profile: null
      query: record_sets
      region: null
      resource_id: null
      security_token: null
      start_record_name: student1-code.win-21820-1.rhdemo.io
      type: A
      validate_certs: true

TASK [code_server : delete dns entries for vs code for each student] **********************************************************
changed: [localhost] => (item=1) => changed=true
  ansible_loop_var: item
  invocation:
    module_args:
      alias: null
      alias_evaluate_target_health: false
      alias_hosted_zone_id: null
      aws_access_key: null
      aws_secret_key: null
      debug_botocore_endpoint_logs: false
      ec2_url: null
      failover: null
      health_check: null
      hosted_zone_id: null
      identifier: null
      overwrite: null
      private_zone: false
      profile: null
      record: student1-code.win-21820-1.rhdemo.io
      region: null
      retry_interval: 500
      security_token: null
      state: absent
      ttl: 3600
      type: A
      validate_certs: true
      value:
      - 3.80.126.3
      vpc_id: null
      wait: false
      wait_timeout: 300
      weight: null
      zone: rhdemo.io
  item: 1

changed: [localhost] => (item=2) => changed=true
  ansible_loop_var: item
  invocation:
    module_args:
      alias: null
      alias_evaluate_target_health: false
      alias_hosted_zone_id: null
      aws_access_key: null
      aws_secret_key: null
      debug_botocore_endpoint_logs: false
      ec2_url: null
      failover: null
      health_check: null
      hosted_zone_id: null
      identifier: null
      overwrite: null
      private_zone: false
      profile: null
      record: student2-code.win-21820-1.rhdemo.io
      region: null
      retry_interval: 500
      security_token: null
      state: absent
      ttl: 3600
      type: A
      validate_certs: true
      value:
      - 34.228.61.238
      vpc_id: null
      wait: false
      wait_timeout: 300
      weight: null
      zone: rhdemo.io
  item: 2
```

**Comment by cloin** [comment]
Tested again with same results.

**Comment by IPvSean** [comment]
@cloin is this fixed now with PR #699
